### ISSUE TITLE
Remote Access through vpn server-> adb device [offline]?

### ISSUE THREAD

**Issue by alan0048** [Genymobile/scrcpy#3848]
- [ ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: [Windows 10 64-bit]
 - scrcpy version: [2.0 win_x64]
 - installation method: [Windows release, cmd]
 - device model: Nvidia Shield Pro
 - Android version: [Android TV]

Using the command line "scrcpy -m800 -b2M --no-audio" I am able to mirror my Nvidia Shield Pro through my computer while I'm on the same LAN. When away from home, I can usually connect to my local lan through a Wireguard Server. Hence, I thought I'd be able to connect as usual but my adb device only shows offline?

Why isn't it working? Is there a way to get around that?
Computer A --> Internet ---> Wireguard Server ---> Router (Computer B) ---> Nvidia Shield Pro [offline]

**Comment by rom1v** [comment]
You can `adb connect` only one adb client per device.

You might be interested in this page: https://github.com/Genymobile/scrcpy/blob/master/doc/tunnels.md

**Comment by alan0048** [comment]
Some Key/fingerprint got installed on my computer after I ran the 
adb kill-server
adb -a nodaemon server start

I did not understand much after that. How do I delete or reverse these actions? It fucked everything up for me :/

**Comment by alan0048** [comment]
Does this mean that the remote computer has to be turned on at all time in order to access the Android device remotley? I thought you could connect to the Android device solemnly from an outside network, that's what I was hoping to achieve. 

Scenario 1: Computer A --> Router A / Local LAN --> Nvidia Shield (local LAN)
Scenario 2: Computer A --> Router B / Internet --> VPN Adress --> Router A / Local LAN --> Nvidia Shield (local LAN)

**Comment by rom1v** [comment]
> Does this mean that the remote computer has to be turned on at all time in order to access the Android device remotley?

Yes, you communicate with the device via the computer where it is connected, not directly.

**Comment by michaelk-realityi** [comment]
@alan0048 
I have found that only one PC can have an active adb connection to the same device.
If your remote pc is the 2nd device connecting it might connect but show offline.
Before leaving the house make sure to run 'adb disconnect 192.168.10.XXX' (this will disconnect the one device) on the computer that is currently connected to your Nvidia Shield. IF your laptop is the one connected at your house and then attempting to connect via VPN you may need to reboot the Nvidia Shield to stop the adb session on the Nvidia Shield.

I was able to just achieve the reverse of your Scenario 2 with an Android 7 device in my office. It VPN'ed into the office and I connected to it. Then a coworker VPN'ed into the office and connected to the Android 7 device across both VPNs. However I had to 'adb disconnect' from it before my coworker was able connect.

My Scenario 1:
Remote Android A --> (Router B/ Internet) --> VPN Address --> (Router A / Local LAN ) <-- Computer A

My Scenario 2:
Remote Android A --> (Router B/ Internet) --> VPN Address --> (Router A / Local LAN ) <-- VPN Address <-- (Router C/ Internet) Remote Computer 2

**Comment by rom1v** [comment]
> I have found that only one PC can have an active adb connection to the same device.

Yes, this is an adb server limitation.
